transaction/extra/src/memo/authenticated_common.rs

```diff
@@ -5,13 +5,76 @@
 //! This validation scheme was proposed for standardization in
 //! mobilecoinfoundation/mcips/pull/4
 
+use crate::SenderMemoCredential;
+use alloc::boxed::Box;
+use core::fmt::Debug;
 use mc_account_keys::{PublicAddress, ShortAddressHash};
 use mc_crypto_keys::{
     CompressedRistrettoPublic, KexReusablePrivate, RistrettoPrivate, RistrettoPublic,
 };
+use mc_transaction_core::NewMemoError;
 use subtle::{Choice, ConstantTimeEq};
 
-use crate::SenderMemoCredential;
+/// A trait for abstracting away signing of 0x0100 and 0x0101 memos.
+pub trait AuthenticatedMemoHmacSigner: Debug {
```

Comment:
❓ what are these changes about? They don't appear to be being used

Reply:
These changes are necessary for the hw memo signing. They should've been in a follow-up PR to make that clearer and make this PR smaller.

The purpose of them is to allow different HMAC signing implementations and the HW wallet solution in full-service uses that to provide a way of signing using the Ledger.